warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1218,18 +1222,21 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
      * @throws InstantiationException
      * @throws IllegalAccessException
      */
-    private void loadDataTypeMetadata(Multimap<String,Type<?>> fieldToDatatypeMap, Set<String> indexedFields, Set<String> normalizedFields,
-                    ShardQueryConfiguration config, boolean reload) throws AccumuloException, AccumuloSecurityException, TableNotFoundException,
-                    ExecutionException, InstantiationException, IllegalAccessException {
+    private void loadDataTypeMetadata(Multimap<String,Type<?>> fieldToDatatypeMap, Set<String> indexedFields, Set<String> reverseIndexedFields,
+                    Set<String> normalizedFields, ShardQueryConfiguration config, boolean reload) throws AccumuloException, AccumuloSecurityException,
+                    TableNotFoundException, ExecutionException, InstantiationException, IllegalAccessException {
         synchronized (dataTypeMap) {
-            if (!reload && (null != cachedIndexedFields && null != indexedFields) && (null != cachedNormalizedFields && null != normalizedFields)) {
+            if (!reload && (null != cachedIndexedFields && null != indexedFields) && (null != cachedNormalizedFields && null != normalizedFields)
+                            && (null != cachedReverseIndexedFields && null != reverseIndexedFields)) {
                 indexedFields.addAll(cachedIndexedFields);
+                reverseIndexedFields.addAll(cachedReverseIndexedFields);
                 normalizedFields.addAll(cachedNormalizedFields);
                 
                 return;
             }
             
             cachedIndexedFields = metadataHelper.getIndexedFields(null);
+            cachedReverseIndexedFields = metadataHelper.getReverseIndexedFields(null);
```

Comment:
I am thinking this should be filtered by datatype here.....  Or we remove this whole caching mechanism altogether.

Reply:
Yeah, I can't really tell what's supposed to be happening in the `DefaultQueryPlanner` where this method is called.